code/game/turfs/simulated/floor/plating/asteroid.dm

```diff
@@ -273,12 +273,14 @@ GLOBAL_LIST_INIT(megafauna_spawn_list, list(/mob/living/simple_animal/hostile/me
 			else //this is not danger, don't spawn a boss, spawn something else
 				randumb = pickweight(mob_spawn_list)
 
-		for(var/mob/living/simple_animal/hostile/H in urange(12,T)) //prevents mob clumps
-			if((ispath(randumb, /mob/living/simple_animal/hostile/megafauna) || ismegafauna(H)) && get_dist(src, H) <= 7)
+		for(var/atom/A in urange(12, T)) //prevents mob clumps
+			if(!ishostile(A) && !istype(A, /obj/structure/spawner))
+				continue
+			if((ispath(randumb, /mob/living/simple_animal/hostile/megafauna) || ismegafauna(A)) && get_dist(src, A) <= 7)
 				return //if there's a megafauna within standard view don't spawn anything at all
-			if(ispath(randumb, /mob/living/simple_animal/hostile/asteroid) || istype(H, /mob/living/simple_animal/hostile/asteroid))
+			if(ispath(randumb, /mob/living/simple_animal/hostile/asteroid) || istype(A, /mob/living/simple_animal/hostile/asteroid))
```

Comment:
is there a reason we're checking for paths? I imagine you can't have a typepath in a turf.

Reply:
I'm not so sure you're reading the code correctly.

the path checking isn't checking for anything on the *turf*, it's checking for what is about to be *spawned*; the process picks a mob path from a path list of mobs; the path checking has nothing to do with the contents of the turf or what's on it.

Reply:
Ah right yeah I see now.